tools/build_node.js

```diff
@@ -6,42 +6,102 @@ const { filter } = require("./lib/dependencies");
 const { rollupWrite } = require("./lib/bundling.js");
 const log = (...args) => console.log(...args);
 
-async function buildNodeIndex(languages) {
+async function buildESMIndex(name, languages) {
+  const header = `import hljs from './core.mjs';`;
+  const footer = "export default hljs;";
+
+  const safeImportName = (s) => {
+    s = s.replace(/-/g, "_");
+    if (/^\d/.test(s)) s = `L_${s}`;
+    return s;
+  };
+
+  const registration = languages.map((lang) => {
+    let out = '';
+    const importName = safeImportName(lang.name);
+    const require = `import ${importName} from './languages/${lang.name}.js';`;
+    out += require;
+    out += `hljs.registerLanguage('${lang.name}', ${importName});`;
+    return out;
+  });
```

Comment:
nit: I think it's actually more readable without the variables
```suggestion
  const registration = languages.map((lang) => 
    `import ${safeImportName(lang.name)} from './languages/${lang.name}.js';` +
    `hljs.registerLanguage('${lang.name}', ${importName});`);
```